### DISCUSSION TITLE
Stopping AR programatically

### DISCUSSION THREAD

**Discussion by benferns** [google/model-viewer#2979] [Q&A]
Hi,

I'd like to be able to stop AR via script, based on some on-screen UI interaction (I can't use the close button slot unfortunately).

I can see the existing method is via https://github.com/google/model-viewer/blob/ef0b6b68a0adb4215f08c9b5059da5638d431a86/packages/model-viewer/src/three-components/ARRenderer.ts#L266 , but is there a more generic  'viewer.stopAR()' or similar somewhere? Could I close the session by manipulating attributes on the model-viewer elements?

**Comment by samaneh-kazemi** [comment]
Hi @timmmeh, do you have any feedback on this?

**Reply by ghost** [reply]
I don't think a feature like this exists but would probably be straightforward to implement (would only work for WebXR of-course).
